src/BinSkim.Rules/DwarfRules/BA3003.EnableStackProtector.cs

```diff
@@ -69,10 +76,15 @@ public override void Analyze(BinaryAnalyzerContext context)
             IDwarfBinary binary = context.DwarfBinary();
             List<DwarfCompileCommandLineInfo> failedList;
 
-            static bool analyze(IDwarfBinary binary, out List<DwarfCompileCommandLineInfo> failedList)
+            static bool analyzeDwarf(IDwarfBinary binary, out List<DwarfCompileCommandLineInfo> failedList)
             {
                 failedList = new List<DwarfCompileCommandLineInfo>();
 
+                if (binary.CommandLineInfos.Count < 1)
```

Comment:
>CommandLineInfos [](http://example.com/codeflow?start=27&length=16)

Curious, why did you add this early exit? Strictly speaking, doesn't look necessary, am I correct on that? I notice you didn't introduce a similar early exit in the other static local function, is there a difference between the two cases?

Reply:
Yeah it's not *really* necessary. I just wanted to be explicit in returning false if there is no DWARF command line information. It could be added to `analyzeSymbols` too without changing the behavior.